RHEcosystemAppEng/dbaas-operator
# [DBAAS-1044] DBaaS Connection Resource Metrics Addition

## Description
DBaaSConnection metrics based on the design proposal at this [link](https://docs.google.com/presentation/d/1R9jdTdBwDp70Xlr9r_83nsbG7JUqA8I0DsnKE_N3hsk/edit?usp=sharing)

Prometheus Output Console
<img width="1213" alt="image" src="https://user-images.githubusercontent.com/87712456/202539384-5863b1b5-3910-4ba9-81c1-3bc3a6b316ae.png">

## Verification Steps on OpenShift Local
* Please check screenshot of the Prometheus output [here](https://docs.google.com/document/d/1EA4U7mFSQHsq8xS-cfvPuQ7BxdUh9ziCE0OZZKF-H_0/edit?usp=sharing)
* On fresh Openshift Local instance, in a terminal window login with admin credentials and deploy DBaaS Operator 

```shell
oc apply -f config/samples/installation-for-non-OSD-OCP-environments.yaml
```
* Once the DBaaS Operator is fully installed, expose the Prometheus service and navigate to the prometheus service in the browser

```shell
oc project openshift-dbaas-operator
oc expose svc/prometheus-operated
oc get route
```

* Deploy Mongo Quick Start sample
```shell
oc create ns dbaas-rhkp
oc project dbaas-rhkp
oc apply -f mongo-quickstart/deployment.yaml 
```

* Follow usual process to add provider such as MongoDB Atlas with Org Id, Public Key and Secret. ( Namespace: openshift-dbaas-operator, Database Services -> Database Access )
* Create the Database Instance from Data Services -> Data Access -> Configuration -> Create Database Instance. Select appropriate Provider Account, Instance & Project Name and create the instance.
* Switch to Developer Perspective -> + Add -> Cloud Hosted Database -> MongoDB Atlas Cloud Database Service -> Add to Topology, Select Instance and hit Add to Topology
* Check the prometheus console with dbaas_requests_duration_seconds_bucket, dbaas_requests_duration_seconds_count, dbaas_requests_duration_seconds_sum metrics with resource: dbaas_connection and event: create.
* You may see metric dbaas_requests_error_count in case if there were errors, just in case.
Developer Perspective -> 3 dots of Connection -> Delete DBaaSConnection, check the prometheuy metrics again with resource: dbaas_connection and event: delete.
* Delete the installed operator with
```shell
oc delete -f config/samples/installation-for-non-OSD-OCP-environments.yaml
```

## Type of change

- [ ] Enhancements to an existing feature

## Checklist (Definition of Done)
